File: anagram.py
# Given two strings, write a method to decide
# if one is a permutation (anagram) of the other



def anagram(s1, s2):
  #can't be anagrams if strings unequal lengths
  if len(s1) != len(s2):
    status = False
  else:
    sorted1 = sorted(s1.lower()) #returns an array
    sorted2 = sorted(s2.lower()) #returns an array
    for i in sorted1:
      for j in sorted2:
        if i == j:
          status = True
        else:
          status = False

  return status #true if anagram, false if not

def main():
  print str(anagram('HELLO', 'eHLlo'))

if __name__ == "__main__":
    main()

File: Anagram.java
import java.util.Arrays;

public class Anagram {


  // sort letters in input string and return sorted string
  public static String sort(String s) {
    char[] letters = s.toCharArray();
    Arrays.sort(letters);
    return new String(letters);
  }

  // returns True if s permutation of t, False otherwise
  public static boolean permutation(String s, String t) {
    return sort(s).equals(sort(t));

  }

  public static boolean anagram(String s, String t) {
    boolean status;
    if (s.length() != t.length()) {
      status = false;
    }
    char[] s_arr = s.toCharArray();
    char[] t_arr = t.toCharArray();

    for (char i : s_arr) {
      for (char j : t_arr) {
        if (i == j) {
          status = true;
        } else {
          status = false;
        }
      }
    }
    return status;
  }


  public static void main(String[] args) {
    String [][] pairs = {{"apple", "papel"}, {"carrot", "tarroc"}, {"hello", "llloh"}};
    for (String[] pair : pairs) {
      String s1 = pair[0];
      String s2 = pair[1];

      //first check: are the strings permutations of one another?
      boolean isAnagram = permutation(s1, s2);
      System.out.println(s1 + ", " + s2 + ": " + isAnagram);

      //second check: are the strings anagrams of one another?
      System.out.println(anagram(s1, s2));

    }

  }

}